source/features/pr-easy-toggle-files.tsx

```diff
@@ -0,0 +1,26 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+function toggleFile(event: Event): void {
+	const elementClicked = event.target as HTMLElement;
+	const headerBar = elementClicked.closest<HTMLElement>('.file-header.file-header--expandable')!;
+
+	// The clicked element is either the bar itself or one of its 2 children
+	if (elementClicked === headerBar || elementClicked.parentElement === headerBar) {
+		select('.file-info > button', headerBar)!.click();
```

Comment:
```suggestion
		select('[aria-label="Toggle diff contents"]', headerBar)!.click();
```

Reply:
Are you sure? I thought relying on an attribute so "unstable"/volatile (I mean his value could change more easily and more quickly than the others) would not be a good idea.

I also thought about using `[aria-expanded]` as in the left part of the bar, only the button would have this attribute.

Reply:
> Are you sure? I thought relying on an attribute so "unstable"/volatile (I mean his value could change more easily and more quickly than the others) would not be a good idea.

heheh welcome to the world of "hacking on content you don't own."

**Anything** we use can change with **any** frequency, so it's best to use the most unique identifiers rather than loose selectors, because GitHub might change the order of elements and suddenly `.file-info > button` does something you don't expect at all.

If `[aria-label="Toggle diff contents"]` changes, the click will do nothing (with a console error) and that's still preferable to "doing the unexpected."

GitHub has been removing many unique classes so we're left with fickle position-based selectors. Where possible, even if it requires more code, it's best to still select the most unique part of the page and then traverse the DOM from there, for example:

https://github.com/sindresorhus/refined-github/blob/38fb7b001222421c6e3d3e17bfddb3d373a37739/source/features/deprioritize-marketplace-link.tsx#L22-L24

`data-ga-click` and `aria-label` are the attributes that are less likely to change, whereas even `.file-info` could be gone one day in place of `.box.position-relative.px-2`

Reply:
And I completely understand, and in this direction I still thought `[aria-expanded]` would be a better choice. But I will update the selector like your suggestion then.